### PR TITLE
fix/Updates user permissions to handle errors

### PR DESCRIPTION
**Pull request by thiessenp-cds**
# Summary | Résumé

Updates user permissions to handle errors.

Example Error:

Screenshot 2024-05-27 at 9 40 40 AM

## Test

Probably the easiest way to test this is to locally checkout this branch and have the action return an error:
-action /app/(gcforms)/[locale]/(app administration)/admin/(with nav)/accounts/[id]/manage-permissions/actions.ts
-example url http://localhost:3000/en/admin/accounts/clwkrfijj000j8evpjmcp8xvs/manage-permissions

### CODE REVIEW

File: app/(gcforms)/[locale]/(app administration)/admin/(with nav)/accounts/[id]/manage-permissions/components/client/Privilege.tsx
Comment (thiessenp-cds) [inline app/(gcforms)/[locale]/(app administration)/admin/(with nav)/accounts/[id]/manage-permissions/components/client/Privilege.tsx:32]: In the future we should decide on a pattern for returning errors from an action. This is a hardcoded error and would break if the server action could return two or more errors.
One idea is to have the server action return the error key.
Future thought :)

File: i18n/translations/fr/admin-users.json
Comment (timarney) [inline i18n/translations/fr/admin-users.json:59]: We should get a translation for this before merging.

Comment (thiessenp-cds) [reply i18n/translations/fr/admin-users.json:59]: Sounds good to me :)
@anikbrazeau would it be possible to get a translation for this string "Failed to update permissions."?

**Comment by timarney** [comment]
@thiessenp-cds --- pushed a tiny commit to update the loading animation colour to use something from the app colour palette  --- ping to to approve once we have the string.

Tested and it's working as expeected.

File: i18n/translations/fr/admin-users.json
Comment (anikbrazeau) [inline i18n/translations/fr/admin-users.json:59]:
```suggestion
    "permissions": "Les autorisations n'ont pas pu être mises à jour."
```

**Review by timarney — APPROVED** [review]

**Pull request merged**
